draft-bbs-signatures.md

```diff
@@ -748,6 +753,81 @@ Procedure:
 5. return (scalar_1, ..., scalar_n)
 ```
 
+### OctetsToSignature
+
+This operation describes how to decode an octet string, validate it and return the underlying components that make up the signature.
+
+```
+(A, e, s) = octets_to_signature(signature_octets)
+
+Inputs:
+
+- signature_octets (REQUIRED), octet string of the form output from signature_to_octets operation.
+
+Outputs:
+
+- A, a valid point in the G1 subgroup which is not equal to the identity point.
+- e, a non-negative integer representing a valid scalar value with the range of 0 < e < q.
+- s, a non-negative integer representing a valid scalar value with the range of 0 < e < q.
+
+Procedure:
+
+1. if len(signature_octets) != (octet_point_length + 2 * octet_scalar_length), return INVALID
+
+2. a_octets = signature_octets[0..(octet_point_length - 1)]
+
+3. A = octets_to_point(a_octets)
+
+4. if A is INVALID, return INVALID
+
+5. if A == Identity_G1, return INVALID
+
+5. index = octet_point_length
+
+6. e = OS2IP(signature_octets[index..(index + octet_scalar_length - 1)])
+
+7. if e >= q, return INVALID
```

Comment:
```suggestion
7. if e = 0 OR e >= q, return INVALID
```
I think we do need to check if e and s are zero right?? @andrewwhitehead??

Reply:
`s` could potentially be zero for an unblinded signature if `s' + s'' = 0`. If an issuer wants to use `s = 0` or `e = 0` then they could, but I don't think it's great from a security perspective

Reply:
Addressed for e and s

Reply:
Although i don't really see any security problems with having either e or s be 0 on the top of my head the paper does define them in the multiplicative group mod q so i would prefer if they where non 0 just for that.

Reply:
The point about blind signatures is also very interesting. If we decide to disallow 0 `s` we may have a problem there (not really saying to change it in the core draft, just that it is something we will have to consider when updating the blind signatures spec).